eo-maven-plugin/src/test/java/org/eolang/maven/optimization/OptCachedTest.java

```diff
@@ -50,73 +49,92 @@ final class OptCachedTest {
     /**
      * Test case for XML program in cache.
      *
-     * @param tmp Temp dir
+     * @param cache Temp cache dir.
+     * @param dir Temp program dir
      * @throws IOException if I/O fails
      * @todo #2422:60min returnsFromCacheIfXmlAlreadyInCache: this test is unstable.
      *  We should resolve issues with unstable failures and only
      *  then enable the test.
      *  Also, see this <a href="https://github.com/objectionary/eo/issues/2727">issue</a>.
+     * @todo #2790:30min There is repeating logic in tests.
+     *  Code logic repeats in {@link OptCacedTest},
+     *  {@link OptimizeMojoTest.getAlreadyOptimizedResultsFromCache}
+     *  and {@link ShakeMojoTest.getAlreadyShakenResultsFromCache}.
+     *  To check whether a program is loaded from the cache,
+     *  different tests use the same code
+     *  { @code Files.setLastModifiedTime(
+     *  res,
+     *  FileTime.fromMillis(System.currentTimeMillis() + time)
+     *  ); }.
+     *  We need to think about how to remove this duplication.
      */
-    @Disabled
     @Test
-    void returnsFromCacheIfXmlAlreadyInCache(@TempDir final Path tmp) throws IOException {
-        final XML program = OptCachedTest.program(ZonedDateTime.now());
-        OptCachedTest.save(tmp, program);
+    void returnsFromCacheIfXmlAlreadyInCache(@TempDir final Path cache, @TempDir final Path dir)
+        throws Exception {
+        final XML xml = OptCachedTest.program();
```

Comment:
@Yanich96, can we save different XML content to the `cache` and `dir` folders? However, the corresponding files must have the same "last modified" time.

Let me explain the reason. As a reader, I see that you have two files in the `dir` and `cache` folders with the same content. Later, you simply compare this content with the result from `OptCached`. So, we can't be sure from which place `OptCache` returns the result – it might be the cache or just the directory. Since the content is the same, we can't be certain.